### PR TITLE
Internal 129 dgraph ops tools (graplctl)

### PR DESCRIPTION
**Pull request by jgrillo-grapl**
### Which issue does this PR correspond to?
https://github.com/grapl-security/issue-tracker/issues/129

### What changes does this PR make to Grapl? Why?

This PR adds a CLI tool `graplctl` which is mean to be used to assist with Grapl operations.

This PR includes the following changes:
1. `graplctl swarm` commands to spin up EC2 instances and provision a Docker Swarm cluster.
2. `graplctl dgraph` commands to deploy DGraph on a Docker Swarm cluster.
3. Remove some Swarm and DGraph resources from CDK, as they are now managed by `graplctl`.
4. Update operational documentation to reflect that `graplctl` should be used for setting up the DGraph database.

### How were these changes tested?
Tested manually in AWS sandbox

### CODE REVIEW

File: src/python/graplctl/graplctl/docker_swarm_ops.py [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+import boto3
+from botocore.exceptions import ClientError
+
+from mypy_boto3_ec2 import EC2ServiceResource
+from mypy_boto3_cloudwatch.client import CloudWatchClient
+from mypy_boto3_cloudwatch.type_defs import MetricTypeDef
+from mypy_boto3_sns.client import SNSClient
+from mypy_boto3_ssm import SSMClient
+
+from . import common
```
Comment (wimax-grapl) [inline src/python/graplctl/graplctl/docker_swarm_ops.py:14]: I don't love `from .` but I'm willing to overlook it for now :)

File: src/python/graplctl/graplctl/cli.py [hunk 2]
```diff
@@ -0,0 +1,607 @@
...
+    envvar="GRAPL_REGION",
+    help="grapl region to target [$GRAPL_REGION]",
+)
+@click.option(
+    "-p",
+    "--grapl-prefix",
+    type=click.STRING,
+    envvar="GRAPL_CDK_DEPLOYMENT_NAME",
+    help="grapl deployment name [$GRAPL_CDK_DEPLOYMENT_NAME]",
+)
```
Comment (wimax-grapl) [inline src/python/graplctl/graplctl/cli.py:59]: nitpick: 
in the spirit of https://github.com/grapl-security/issue-tracker/issues/125 could we maybe call this `--grapl-deployment-name` / `-n` or something?

(both verbiages are used here)

File: src/python/graplctl/graplctl/cli.py [hunk 3]
```diff
@@ -0,0 +1,612 @@
...
+@click.option(
+    "-i",
+    "--swarm-id",
+    type=click.STRING,
+    help="unique ID for this swarm cluster (random default)",
+    default=str(uuid.uuid4()),
+)
+@click.pass_obj
+def create(
```
Comment (wimax-grapl) [inline src/python/graplctl/graplctl/cli.py:114]: sorry, here would be where we shoudl validate instance type.
also, thoughts on maybe some sort of preventative mechanism like "you've already `create`'d for this deployment" or something?

**Review by ajm-grapl — APPROVED** [review]
LGTM, would you add some usage instructions to the repo, perhaps to the readme?

**Review by christophermaier — COMMENTED** [review]
Always a fan of better automation!

I'm curious why there are a bunch of Python commands in the `src/js/grapl-cdk` tree, though. What's the distinction between them and the new `graplctl` tool?

**Comment by jgrillo-grapl** [comment]
> I'm curious why there are a bunch of Python commands in the `src/js/grapl-cdk` tree, though. What's the distinction between them and the new `graplctl` tool?

@christophermaier those scripts get deployed by the CDK to an S3 bucket, then pulled down to the instances by the SSM agent and run when we send a RunCommand invocation via the CLI. They're basically the "remote" component of each command.

File: src/python/graplctl/graplctl/docker_swarm_ops.py (hunk 1 above)
Comment (jgrillo-grapl) [reply src/python/graplctl/graplctl/docker_swarm_ops.py:16]: Likewise :p

**Pull request merged**